### PR TITLE
Add pagination

### LINKED ISSUE #68: Add Pagination

### PR DESCRIPTION
**Pull request by bexrr**
closes #68

### CODE REVIEW

**Review by Agreon — APPROVED** [review]
Wow :D

File: server/income.go [hunk 1]
```diff
@@ -44,7 +45,14 @@ func listIncomes(c *gin.Context) {
...
 	userID := c.MustGet("userID")
 
-	err := db.Select(&income, "SELECT * FROM income WHERE user_id=$1 AND active=$2 ORDER BY created_at DESC", userID, true)
+	page := c.Query("page")
+	pageInt, err := strconv.Atoi(page)
+	if err != nil {
+		saveErrorInfo(c, err, 400)
+		return
+	}
+	page = strconv.Itoa(pageInt * pageSize)
```
Comment (Agreon) [inline server/income.go:54]: Could you extract that into a method/middleware?

**Review by Agreon — CHANGES_REQUESTED** [review]

File: budgie-app/src/screens/expenses/CreateExpense.tsx [hunk 2]
```diff
@@ -5,26 +5,29 @@ import { View, SafeAreaView } from 'react-native';
...
   const { showToast } = useToast();
+  const queryClient = useQueryClient();
 
   const [availableTags, setAvailableTags] = useState<Tag[] | null>(null);
 
   useEffect(() => {
     (async () => {
       try {
-        const { data: tags } = await api.get('tag');
+        const { data: tags } = await api.get('tag?page=0');
```
Comment (bexrr) [inline budgie-app/src/screens/expenses/CreateExpense.tsx:30]: What's that supposed to be?????? :rage:

File: budgie-app/src/screens/expenses/EditExpense.tsx [hunk 3]
```diff
@@ -36,7 +39,7 @@ export const EditExpense: FC<{
     (async () => {
       try {
         const { data } = await api.get(`expense/${id}`);
-        const { data: tags } = await api.get('tag');
+        const { data: tags } = await api.get('tag?page=0');
```
Comment (bexrr) [inline budgie-app/src/screens/expenses/EditExpense.tsx:42]: Same as above..

**Pull request merged**